src/test/java/com/amihaiemil/docker/RtContainerTestCase.java

```diff
@@ -55,4 +60,116 @@ public void returnsId() {
+                        .add("Name", "boring_euclid")
+                        .build().toString()
+                ),
+                new Condition(
+                    "Method should be a GET",
+                    req -> req.getRequestLine().getMethod().equals("GET")
+                ),
+                new Condition(
+                    "Resource path must be /{id}/json",
+                    req -> req.getRequestLine().getUri().endsWith("/123/json")
+                )
+            ),
+            URI.create("http://localhost:80/1.30/containers/123")
+        );
+        final JsonObject info = container.inspect();
+        MatcherAssert.assertThat(info.keySet(), Matchers.hasSize(3));
+        MatcherAssert.assertThat(
+            info.getString("Id"), Matchers.equalTo("123")
+        );
+        MatcherAssert.assertThat(
+            info.getString("Image"), Matchers.equalTo("some/image")
+        );
+        MatcherAssert.assertThat(
+            info.getString("Name"), Matchers.equalTo("boring_euclid")
+        );
+    }
+
+    /**
+     * RtContainer.inspect() returns null because the HTTP Response's status
+     * is not 200 OK.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public void inspectsNotFound() throws Exception {
+        final Container container = new RtContainer(
+            new AssertRequest(
+                new Response(HttpStatus.SC_NOT_FOUND, ""),
+                new Condition(
+                    "Method should be a GET",
+                    req -> req.getRequestLine().getMethod().equals("GET")
+                ),
+                new Condition(
+                    "Resource path must be /{id}/json",
+                    req -> req.getRequestLine().getUri().endsWith("/123/json")
+                )
+            ),
+            URI.create("http://localhost:80/1.30/containers/123")
+        );
+        MatcherAssert.assertThat(container.inspect(), Matchers.nullValue());
+    }
+
+    /**
+     * RtContainer can start with no problem.
+     * @throws Exception If something goes wrong.
+     */
+    @Test
+    public void startsOk() throws Exception {
+        new RtContainer(
+            new AssertRequest(
+                new Response(
+                    HttpStatus.SC_NO_CONTENT, ""
+                ),
+                new Condition(
+                    "Method should be a POST",
+                    req -> req.getRequestLine().getMethod().equals("POST")
+                ),
+                new Condition(
+                    "Resource path must be /{id}/start",
+                    req -> req.getRequestLine().getUri().endsWith("/123/start")
+                )
+            ),
+            URI.create("http://localhost:80/1.30/containers/123")
+        ).start();
+    }
+
+    /**
+     * RtContainer throws ISE if it cannot start.
+     * @throws Exception If something goes wrong.
+     */
+    @Test(expected = IllegalStateException.class)
```

Comment:
@amihaiemil `RtContainer.start()` missing tests on receiving `404` and `304` from the docker API